### ISSUE TITLE
How "ignore_mask", with which we only pick the "pure background" in grid cells

### ISSUE THREAD

**Issue by ShihuaiXu** [david8862/keras-YOLOv3-model-set#196]
非常感谢david的代码，我在模型训练过程中，将class得分和conf得分做比较时，发现conf的预测比较不准，导致模型的精度受限，我阅读了之前一位朋友的issue，地址https://github.com/david8862/keras-YOLOv3-model-set/issues/124
对confidence的loss的nonobj部分有一些疑问，
(1-object_mask) * K.binary_crossentropy(object_mask, raw_pred[...,4:5], from_logits=True)计算了gt中没有目标的点的交叉熵损失，后面乘上了ignore_mask，按理说这个ignore_mask的作用是对背景的筛选，但我发现ignore_mask的计算过程是以iou为标准的，此时我产生了一个疑问，既然是背景，那么这个位置肯定没有GT的标注，也就是没有bbox1，bboy1，bbox2，bboxy2这些点，那么如何计算iou呢，我对ignore_mask的计算过程比较迷惑，麻烦david看到帮忙解答一下
thanks in advance

**Comment by david8862** [comment]
@ShihuaiXu  目前的实现里背景anchor的GT中的(x,y,w,h)会使用默认的0值，因而与相应的pred计算得到的iou也是0

**Comment by ShihuaiXu** [comment]
@david8862 是的，我又看了其他资料，发现在13*13的map中，除了GT中的中心点的那些cell，其他点也可以预测GT中的那些bbox，虽然中心点不在GT的cell上，可能有些附件的cell点pred出来的box和GT的iou比较大，因为bbox长和宽可以比较大。

想问下david，目前我的模型的分类和定位的表现都还不错，就是这个置信度不大准，想问下有哪些方法可以改进，比如说加大confidence loss的权值？具体我该如何实现呢

**Comment by david8862** [comment]
在YOLOv5的loss计算中对于三个部分的loss是设定了相应的比例系数的，可以在[这里](https://github.com/david8862/keras-YOLOv3-model-set/blob/master/yolo3/loss.py#L364)求和时设置

**Comment by ShihuaiXu** [comment]
非常感谢，我再尝试一下
